### ISSUE TITLE
The resource 'hsdp_iam_group_membership' is not removing user membership from IAM on destroy

### ISSUE THREAD

**Issue by l-lafin** [philips-software/terraform-provider-hsdp#311]
The resource 'hsdp_iam_group_membership' is not removing user membership from IAM on destroy nor when removing the user from the list, but it also doesn't fail the terraform run and claim a successful execution.

Version: 0.41.0
![image](https://user-images.githubusercontent.com/47752087/216451782-aac71bf0-b22a-42d4-915b-9d9875703c9f.png)

**Comment by loafoe** [comment]
@l-lafin added additional test code and was able to observe deletion of users group on destroy. On key thing is that groups referred to in `hsdp_iam_group_membership` should have `drift_detection = true` otherwise you get permadiffs, or possibly the issue you are seeing as mentioned in this issue. Adding `hsdp_iam_group_membership` in retrospect was a mistake IMHO. It goes against the ownership of resources and feels more and more like a footgun. Closing

**Comment by l-lafin** [comment]
Hi @loafoe,
Thanks for the updates, regarding the `drift_detection` according to the documentation in case the group is managed by Terraform. it indicates the  `drift_detection` should be `false`, see below:

![image](https://user-images.githubusercontent.com/47752087/234412713-39b161d8-840c-437c-8106-8d11e870def8.png)

We are still trying to fix this issue on our side :(, I tried to simulate using plain Terraform and indeed I wasn't able to see the error, but for some reason when I'm using Terragrunt I'm having this error.

**Comment by loafoe** [comment]
You are right, it should be set to `false` , somehow inverted that in my msg🤦‍♂️

I'm not a fan of terragrunt. It feels like what coffeescript was to javascript i.e. plastering over some imperfections and making  things more opaque but, by design, also making runs less transparent..

**Comment by l-lafin** [comment]
Hi @loafoe, I just discovered what was happening. The issue happens when the identity (service/user) doesn't have the permission `HSDP_IAM_ORGANIZATION.MGMT` on that specific IAM organization, so it is an authorization issue. That being said it's weird that the provider is not displaying a `403` error when it is not able to remove the user from the group.

I also dived even further and took a look into the provider code and it seems to be calling the `$remove-members` API, but according to the documentation (see image), any identity with `HSDP_IAM_ORGANIZATION.MGMT`, `GROUP.WRITE` or `HSDP_IAM_GROUP.REMOVE_USER` should be able to remove the members...but in reality, only identities with `HSDP_IAM_ORGANIZATION.MGMT` are able to remove members.

![image](https://user-images.githubusercontent.com/47752087/234660619-92c1787a-1cd9-406d-a84e-85f9c510540f.png)

**Comment by l-lafin** [comment]
We also tested the IAM API itself and it works when the identity has any of those permissions.
